addons/rearm/functions/fnc_rearmSuccess.sqf

```diff
@@ -21,23 +21,21 @@
  */
 #include "script_component.hpp"
 
-private ["_dummy", "_weaponSelect", "_turretOwnerID"];
 params [["_args", [objNull, objNull, [], 0, "", 0], [[]], [6]]];
-_args params ["_target", "_unit", "_turretPath", "_numMagazines", "_magazineClass", "_numRounds"];
+_args params ["_vehicle", "_unit", "_turretPath", "_numMagazines", "_magazineClass", "_numRounds"];
 
-//hint format ["Target: %1\nTurretPath: %2\nNumMagazines: %3\nMagazine: %4\nNumRounds: %5", _target, _turretPath, _numMagazines, _magazineClass, _numRounds];
+//hint format ["Vehicle: %1\nTurretPath: %2\nNumMagazines: %3\nMagazine: %4\nNumRounds: %5", _vehicle, _turretPath, _numMagazines, _magazineClass, _numRounds];
 
 if (local _unit) then {
     [_unit, true, true] call FUNC(dropAmmo);
 };
 
 if (isServer) then {
-    _turretOwnerID = _target turretOwner _turretPath;
+    private _turretOwnerID = _vehicle turretOwner _turretPath;
     if (_turretOwnerID == 0) then {
-        [_this, QFUNC(rearmSuccessLocal), _target] call EFUNC(common,execRemoteFnc);
+        [QGVAR(rearmSuccessLocalEH), _vehicle, _this] call EFUNC(common,objectEvent);
     } else {
-        EGVAR(common,remoteFnc) = [_this, QFUNC(rearmSuccessLocal), 0];
-        _turretOwnerID publicVariableClient QEGVAR(common,remoteFnc);
+        [QGVAR(rearmSuccessLocalEH), _turretOwnerID, _this] call EFUNC(common,targetEvent);
     };
 } else {
     [_this, QFUNC(rearmSuccess), 1] call EFUNC(common,execRemoteFnc);
```

Comment:
As discussed on Slack, use `EFUNC(common,serverEvent)` here and in postInit add the event in `isServer` only.